static/js/click_handlers.js

```diff
@@ -278,6 +282,35 @@ exports.initialize = function () {
 
     // MESSAGE EDITING
 
+    $("#main_div").on("mouseenter", ".messagebox", function (e) {
+        e.stopPropagation();
+
+        const message_row = $(this).closest(".message_row");
+        const id = rows.id(message_row);
+        const message = current_msg_list.get(id);
+        const now = new XDate();
+        const seconds_left =
+            page_params.realm_message_content_edit_limit_seconds +
+            now.diffSeconds(message.timestamp * 1000);
+        const is_message_editable =
+            message_edit.get_editability(message) === message_edit.editability_types.FULL;
+
+        if (is_message_editable) {
+            timeout = setTimeout(() => {
+                const args = {
+                    is_editable: false,
+                    msg_id: id,
+                };
+                message_row.find(".edit_content").html(render_edit_content_button(args));
+            }, seconds_left * 1000);
+        }
+    });
```

Comment:
It seems expensive to start a timer every time the mouse enters a messagebox; how did you think about whether that's the right way to achieve our goals here?

Reply:
I would have thought a simpler approach would be to start a timer when we open the `message_edit` UI, instead.

Reply:
I think the `message edit` UI has its own timer that gives the time left at the bottom to edit. For the message row which is in focus, I think we can add various conditions whether to start a timer or not. That way it will not be much expensive I suppose, and also have the right behavior of changing the icon if still in focus.

Reply:
> It seems expensive to start a timer every time the mouse enters a messagebox; how did you think about whether that's the right way to achieve our goals here?



@timabbott, If I am not wrong, the timer is triggered only when the message (upon which the user is currently hovering) is editable or has a `editability_types` as `FULL`. This is only the case when the message timer is within its realm deadline and it is send by the current user. (As verified by the <a href ='https://github.com/zulip/zulip/blob/caa7548c4a24cf9e87471e23d72bd19f9f9527e1/static/js/message_edit.js#L62'>following function</a>).

Would love to know If I am missing out something :)